# RNA-Seq
<p align="justify">

  
## QC and Read mapping in Rice
To perform QC on the rawdata fasq files and remove low qality reads and adaptor sequences etc. Then to mapped the raw reads to reference genome, sort and index the bam file (mapped and unmampped reads)

### Datasets
1. Raw data file (R1.fastq.gz and R2.fastq.gz): These are the file from yout own experiments or can be downloaded from publical repositories like NCBI
2.  [The transcript annotation file (all.gff3)](http://rice.uga.edu/pub/data/Eukaryotic_Projects/o_sativa/annotation_dbs/pseudomolecules/version_7.0/all.dir) This file is used when you are mapping with respect to rice MSU locus ID



